src/checked_transaction.rs

```diff
@@ -177,12 +263,48 @@ impl AsRef<Transaction> for CheckedTransaction {
     }
 }
 
+impl AsMut<Transaction> for CheckedTransaction {
```

Comment:
CheckedTransaction should be immutable / all state changes to the transaction should be mediated by the CheckedTransaction wrapper. Providing a mutable reference to the transaction opens a backdoor that allows the transaction to be mutated in an invalid way, thus violating the contract of CheckedTransaction. 

Instead, we should either:

1. move any mutations to safe methods managed by checked transaction
2. convert CheckedTransaction into a tx, then recheck after making the changes.

Reply:
If was set as convenience since this is a new type and we might disrupt the dependencies downstream. Moved to feature `internals`:

https://github.com/FuelLabs/fuel-tx/pull/162/commits/c38733a2abcacef76b78386f1dc0c945c98e963e